lib/Runtime/Language/InlineCache.inl

```diff
@@ -35,9 +36,17 @@ namespace Js
         if (CheckLocal && type == u.local.type)
         {
             Assert(propertyObject->GetScriptContext() == requestContext); // we never cache a type from another script context
-            *propertyValue = DynamicObject::UnsafeFromVar(propertyObject)->GetInlineSlot(u.local.slotIndex);
-            Assert(*propertyValue == JavascriptOperators::GetProperty(propertyObject, propertyId, requestContext) ||
-                (RootObjectBase::Is(propertyObject) && *propertyValue == JavascriptOperators::GetRootProperty(propertyObject, propertyId, requestContext)));
+            if (OutputExistence)
+            {
+                *propertyValue = requestContext->GetLibrary()->GetTrue();
+                Assert(JavascriptOperators::HasProperty(propertyObject, propertyId));
+            }
+            else
```

Comment:
Could be done once rather than inside each "if"? (inside the "ifs" only check !OutputExistence)

Reply:
I'd love to reduce the repetition in this method, but it's different on different paths. The last two, for the Missing case, set propertyValue to false, and if we miss the cache entirely we don't set propertyValue.

---
In reply to: [175172244](https://github.com/Microsoft/ChakraCore/pull/4834#discussion_r175172244) [](ancestors = 175172244)

Reply:
Ah, it's hard to notice, that the last two are different.

---
In reply to: [175178188](https://github.com/Microsoft/ChakraCore/pull/4834#discussion_r175178188) [](ancestors = 175178188,175172244)

Reply:
I also agree -- it would be awesome if this could be refactored into a helper method, but it does seem like it might not save much